### PR TITLE
[DO - 1720] Create PEP03 scoring rule

### PR DESCRIPTION
**Pull request by Andrii17s**

### CODE REVIEW

File: business_register/pep_scoring/rules.py [hunk 1]
```diff
@@ -86,11 +86,12 @@ def calculate_weight(self) -> tuple[int or float, dict]:
             property__valuation__isnull=True,
             type=Property.SUMMER_HOUSE,
             acquisition_date__year__gte=2015,
+            property__declaration__year=self.declaration.year,
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:101]: Please, explain why type is only Property.SUMMER_HOUSE?! We do have:     
    ```
    HOUSE = 1
    SUMMER_HOUSE = 2
    APARTMENT = 3
    ROOM = 4
    GARAGE = 5
    UNFINISHED_CONSTRUCTION = 6
    OFFICE = 8
    OTHER = 10

```

File: business_register/pep_scoring/rules.py [hunk 2]
```diff
@@ -134,11 +135,12 @@ def calculate_weight(self) -> tuple[int or float, dict]:
             property__valuation__isnull=True,
             type=Property.LAND,
             acquisition_date__year__gte=2015,
+            property__declaration__year=self.declaration.year,
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:138]: We have no need to filter year. You can get all data from the declaration that we check

**Review by IegorSoboliev — CHANGES_REQUESTED** [review]

File: business_register/pep_scoring/rules.py (hunk 2 above)
Comment (Andrii17s) [reply business_register/pep_scoring/rules.py:138]: We also check family members declarations, and should filter it by the year.

File: business_register/pep_scoring/rules.py (hunk 1 above)
Comment (Andrii17s) [reply business_register/pep_scoring/rules.py:101]: "Real estate is property consisting of land and the buildings on it", nothing else except SUMMER_HOUSE didn't look suitable for me

File: business_register/pep_scoring/rules.py (hunk 2 above)
Comment (IegorSoboliev) [reply business_register/pep_scoring/rules.py:138]: Family members property should always be in PEP`s declaration

File: business_register/pep_scoring/rules.py (hunk 1 above)
Comment (IegorSoboliev) [reply business_register/pep_scoring/rules.py:101]: As far as I understand al these types should be included in the rule

File: business_register/pep_scoring/rules.py [hunk 3]
```diff
@@ -85,24 +85,22 @@ class IsRealEstateWithoutValue(BaseScoringRule):
...
-        family_ids.append(self.pep.id)
+        property_types = [Property.SUMMER_HOUSE, Property.HOUSE, Property.APARTMENT, Property.ROOM,
+                          Property.GARAGE, Property.UNFINISHED_CONSTRUCTION, Property.OTHER, Property.OFFICE]
         have_weight = PropertyRight.objects.filter(
-            pep_id__in=family_ids,
+            property__declaration_id=self.declaration.id,
             property__valuation__isnull=True,
-            type=Property.SUMMER_HOUSE,
+            type__in=property_types,
             acquisition_date__year__gte=2015,
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:98]: Please, take into consideration here and in other rules that property can be rented. It is described in the right`s type:
```
        (OWNERSHIP, 'Ownership'),
        (BENEFICIAL_OWNERSHIP, 'Beneficial ownership'),
        (JOINT_OWNERSHIP, 'Joint ownership'),
        (COMMON_PROPERTY, 'Common property'),
        (RENT, 'Rent'),
        (USAGE, 'Usage'),
        (OTHER_USAGE_RIGHT, 'Other right of usage'),
        (OWNER_IS_ANOTHER_PERSON, 'Owner is another person'),
        (NO_INFO_FROM_FAMILY_MEMBER, 'Family member did not provide the information'),

```

**Review by IegorSoboliev — APPROVED** [review]

File: business_register/pep_scoring/rules.py [hunk 4]
```diff
@@ -85,24 +85,25 @@ class IsRealEstateWithoutValue(BaseScoringRule):
...
+        property_count = serializers.IntegerField(min_value=0, required=True)
         declaration_id = serializers.IntegerField(min_value=0, required=True)
 
     def calculate_weight(self) -> tuple[int or float, dict]:
-        family_ids = self.pep.related_persons.filter(
-            to_person_links__category=RelatedPersonsLink.FAMILY,
-        ).values_list('id', flat=True)[::1]
-        family_ids.append(self.pep.id)
+        right_types = [PropertyRight.OWNERSHIP, PropertyRight.BENEFICIAL_OWNERSHIP, PropertyRight.JOINT_OWNERSHIP,
+                       PropertyRight.COMMON_PROPERTY, PropertyRight.USAGE, PropertyRight.OTHER_USAGE_RIGHT]
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:93]: definitely here shouldn`t be PropertyRight.USAGE, PropertyRight.OTHER_USAGE_RIGHT. Also, check, please COMMON_PROPERTY

Comment (IegorSoboliev) [reply business_register/pep_scoring/rules.py:93]: Discuss this Nastya, please

Comment (IegorSoboliev) [reply business_register/pep_scoring/rules.py:93]: And also take into consideration that any object can be owned by another person

**Review by IegorSoboliev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**